src/common/src/util/memcmp_encoding.rs

```diff
@@ -170,6 +170,11 @@ fn calculate_encoded_size_inner(
             DataType::Varchar => deserializer.skip_bytes()?,
             DataType::Bytea => deserializer.skip_bytes()?,
             DataType::Int256 => Int256::MEMCMP_ENCODED_SIZE,
+            DataType::Map(_) => {
+                // Map should not be used as key.
+                // This should be banned in frontend and this branch should actually be unreachable.
+                return Err(memcomparable::Error::NotSupported("map"));
```

Comment:
Kindly ask @xiangjinwu to share some ideas.

Reply:
#7981